src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/planned"
+
+module Y2Storage
+  module BootRequirementsStrategies
+    # Auxiliary class that takes information from several sources (current
+    # devicegraph, already planned devices and user input) and provides useful
+    # information (regarding calculation of boot requirements) about the
+    # expected final system.
+    class Analyzer
+      # Constructor
+      #
+      # @param devicegraph     [Devicegraph] starting situation.
+      # @param planned_devices [Array<Planned::Device>] devices that are already planned to be
+      #   added to the starting devicegraph.
+      # @param boot_disk_name  [String, nil] device name of the disk that the system will try to
+      #   boot first. Only useful in some scenarios like legacy boot.
```

Comment:
Yes, it's documented here https://github.com/yast/yast-storage-ng/pull/259/files#diff-8929af5ad49feaa141b33ec8c31010adR56

So a `@see` should be enough